examples/gatsby/src/components/CollectionGrid/CollectionGrid.js

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+import $nacelle from 'services/nacelle';
+
+import ProductGallery from 'components/ProductGallery';
+
+const CollectionGrid = ({ fields }) => {
+  const [products, setProducts] = useState([]);
+  const collectionHandle = fields.collectionHandle;
+
+  useEffect(() => {
+    const fetchCollection = async () => {
+      try {
+        const collection = await $nacelle.data.collection({
+          handle: collectionHandle
+        });
+        const products = await $nacelle.data.products({
+          handles: collection.productLists.find(
+            (listEntry) => listEntry.slug === 'default'
+          ).handles
+        });
+        setProducts(products);
+      } catch {
+        console.warn(`Collection not found with handle: '${collectionHandle}'`);
```

Comment:
Theoretically this could be an issue with `$nacelle.data.products` as well, right? Am wondering if we want to make the error more generic and use the error message from the SDK? Unsure if that's less useful or not.

Reply:
Good point! I've added some more explicit error handling.